server/chainquery/models/ClaimModel.js

```diff
@@ -0,0 +1,177 @@
+const logger = require('winston');
+
+const {
+  assetDefaults: { thumbnail: defaultThumbnail },
+  details: { host }
+} = require('../../../config/siteConfig'); // TODO: Change back to '@config/siteConfig' when done testing
+
+const getterMethods = {
+  generated_extension() {
+    switch (this.content_type) {
+      case 'image/jpeg':
+      case 'image/jpg':
+        return 'jpg';
+      case 'image/png':
+        return 'png';
+      case 'image/gif':
+        return 'gif';
+      case 'video/mp4':
+        return 'mp4';
+      default:
+        logger.debug('setting unknown file type as file extension jpg');
```

Comment:
Is jpg really a good default? Why not blank or unknown?

Reply:
Also how do you handle nulls? A channel claim is not going to have any of the meta data fields. They will be null in the DB.

Reply:
This is the same implementation as existing code in Spee.ch; it's implemented as a generated field.
I may need to revisit this for reusability.